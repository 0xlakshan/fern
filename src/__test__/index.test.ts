import { convertToPostmanCollection } from "../index";

describe("index", () => {
    it("dummy", () => {
        convertToPostmanCollection({
            types: [],
            services: {
                http: [],
                websocket: [],
            },
            errors: [],
        });
    });
});
