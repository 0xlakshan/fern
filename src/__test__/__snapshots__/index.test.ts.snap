// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open API Conversion Blog Post API 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "Author": Object {
        "description": undefined,
        "oneOf": Array [
          Object {
            "properties": Object {
              "_type": Object {},
            },
            "type": "object",
          },
          Object {
            "properties": Object {
              "_type": Object {
                "type": "string",
              },
            },
            "type": "object",
          },
        ],
      },
      "BlogPost": Object {
        "description": "A blog post",
        "properties": Object {
          "author": Object {
            "$ref": "#/components/schemas/Author",
            "description": undefined,
          },
          "content": Object {
            "description": undefined,
            "type": "string",
          },
          "id": Object {
            "$ref": "#/components/schemas/PostId",
            "description": undefined,
          },
          "title": Object {
            "description": undefined,
            "type": "string",
          },
          "type": Object {
            "$ref": "#/components/schemas/PostType",
            "description": undefined,
          },
        },
        "required": Array [
          "id",
          "type",
          "title",
          "author",
          "content",
        ],
        "type": "object",
      },
      "CreatePostRequest": Object {
        "description": undefined,
        "properties": Object {
          "author": Object {
            "$ref": "#/components/schemas/Author",
            "description": "Author is the writer.",
          },
          "content": Object {
            "description": undefined,
            "type": "string",
          },
          "postType": Object {
            "$ref": "#/components/schemas/PostType",
            "description": undefined,
          },
          "title": Object {
            "description": undefined,
            "type": "string",
          },
        },
        "required": Array [
          "title",
          "author",
          "content",
          "postType",
        ],
        "type": "object",
      },
      "PostId": Object {
        "description": undefined,
        "type": "string",
      },
      "PostNotFoundError": Object {
        "description": undefined,
        "properties": Object {
          "id": Object {
            "$ref": "#/components/schemas/PostId",
            "description": undefined,
          },
        },
        "required": Array [
          "id",
        ],
        "type": "object",
      },
      "PostType": Object {
        "description": undefined,
        "enum": Array [
          "LONG",
          "SHORT",
        ],
        "type": "string",
      },
    },
  },
  "info": Object {
    "title": "BlogPost API",
    "version": "0.0.0",
  },
  "openapi": "3.0.1",
  "paths": Object {
    "posts-service/create": Object {
      "post": Object {
        "description": "Creates a blog post and returns blog post id.",
        "operationId": "createPost",
        "parameters": Array [
          Object {
            "description": "I'm a header!",
            "in": "header",
            "name": "x-endpoint-header",
            "required": false,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/CreatePostRequest",
              },
            },
          },
          "description": undefined,
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/PostId",
                },
              },
            },
            "description": "",
          },
        },
        "tags": Array [
          "PostsService",
        ],
      },
    },
    "posts-service/{postId}": Object {
      "get": Object {
        "description": "Gets a blog post.",
        "operationId": "getPost",
        "parameters": Array [
          Object {
            "description": undefined,
            "in": "path",
            "name": "postId",
            "required": true,
            "schema": Object {
              "$ref": "#/components/schemas/PostId",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/BlogPost",
                },
              },
            },
            "description": "",
          },
          "400": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "oneOf": Array [
                    Object {
                      "allOf": Array [
                        Object {
                          "properties": Object {
                            "_error": Object {
                              "enum": Array [
                                "PostNotFoundError",
                              ],
                              "type": "string",
                            },
                          },
                          "type": "object",
                        },
                        Object {
                          "$ref": "#/components/schemas/PostNotFoundError",
                        },
                      ],
                      "description": undefined,
                      "type": "object",
                    },
                  ],
                },
              },
            },
            "description": "",
          },
        },
        "tags": Array [
          "PostsService",
        ],
      },
    },
  },
}
`;
